groovy-language-server/src/main/java/com/palantir/groovylanguageserver/GroovyTextDocumentService.java

```diff
@@ -91,7 +92,9 @@ public GroovyTextDocumentService(CompilerWrapperProvider provider) {
 
     @Override
     public CompletableFuture<List<? extends Location>> references(ReferenceParams params) {
-        throw new UnsupportedOperationException();
+        return CompletableFuture
+                .completedFuture(provider.get().findReferences(params).stream().map(symbol -> symbol.getLocation())
+                        .filter(location -> Ranges.isValid(location.getRange())).collect(Collectors.toList()));
```

Comment:
I'm imaging for a line looks like `Animal friend = Animal.CAT`, when users want to find references of `Animal`, it should be the `Animal` in this line that's being highlighted. But right now seems the `friend` is being highlighted.